tests/components/airzone/test_config_flow.py

```diff
@@ -145,3 +157,163 @@ async def test_connection_error(hass: HomeAssistant):
+
+
+async def test_dhcp_flow(hass: HomeAssistant) -> None:
+    """Test that DHCP discovery works."""
+
+    with patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_version",
+        return_value=HVAC_VERSION_MOCK,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=DHCP_SERVICE_INFO,
+            context={"source": config_entries.SOURCE_DHCP},
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "discovered_connection"
+
+    with patch(
+        "homeassistant.components.airzone.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_hvac",
+        return_value=HVAC_MOCK,
+    ), patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_hvac_systems",
+        side_effect=SystemOutOfRange,
+    ), patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_webserver",
+        return_value=HVAC_WEBSERVER_MOCK,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_PORT: TEST_PORT,
+            },
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == {
+        CONF_HOST: TEST_IP,
+        CONF_PORT: TEST_PORT,
+    }
+
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_dhcp_flow_error(hass: HomeAssistant) -> None:
+    """Test that DHCP discovery fails."""
+
+    with patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_version",
+        side_effect=AirzoneError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=DHCP_SERVICE_INFO,
+            context={"source": config_entries.SOURCE_DHCP},
+        )
+
+    assert result["type"] == data_entry_flow.FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
+
+
+async def test_dhcp_connection_error(hass: HomeAssistant):
+    """Test DHCP connection to host error."""
+
+    with patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.get_version",
+        return_value=HVAC_VERSION_MOCK,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            data=DHCP_SERVICE_INFO,
+            context={"source": config_entries.SOURCE_DHCP},
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "discovered_connection"
+
+    with patch(
+        "homeassistant.components.airzone.AirzoneLocalApi.validate",
+        side_effect=AirzoneError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_PORT: TEST_PORT,
+            },
+        )
+
+        assert result["errors"] == {"base": "cannot_connect"}
+
```

Comment:
We should finish the flow until it gets to a final result here and assert the result type is either an abort or create

Reply:
Should I call `hass.config_entries.flow.async_abort` to get to a final result?
I mean, this test isn't getting to a final result because the user is allowed to change the port and try again...

Reply:
I'd feed it a good port to verify it gets all the way though, the re-attempt works correctly, and the resulting config entry is what is expected

Reply:
> I'd feed it a good port to verify it gets all the way though, the re-attempt works correctly, and the resulting config entry is what is expected

Done in f3e80e85170ae3fa047774ffdfd73a0659dc67be